ecosystem/typescript/sdk/src/bcs/helper.ts

```diff
@@ -45,6 +45,44 @@ export function deserializeVector(deserializer: Deserializer, cls: any): any[] {
   return list;
 }
 
+/**
+ * Serializes a 2D vector with specified item serialization function.
+ * Very dynamic function and bypasses static typechecking.
+ */
+export function serialization2DVectorWithFunc(value: any[][], func: string): Bytes {
```

Comment:
can you generalize this to make it support vectors of 1D, 2D, 3D...?

Reply:
of course.

Reply:
now use `serializeNDimensionalArrayWithFunc`